packages/plugins/documentation/server/services/helpers/build-api-endpoint-path.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+const _ = require('lodash');
+const pathToRegexp = require('path-to-regexp');
+
+const pascalCase = require('./utils/pascal-case');
+const queryParams = require('./utils/query-params');
+const loopContentTypeNames = require('./utils/loop-content-type-names');
+const getApiResponses = require('./utils/get-api-responses');
+
+/**
+ * @description Parses a route with ':variable'
+ *
+ * @param {string} routePath - The route's path property
+ * @returns {string}
+ */
+const parsePathWithVariables = routePath => {
+  return pathToRegexp
+    .parse(routePath)
+    .map(token => {
+      if (_.isObject(token)) {
+        return token.prefix + '{' + token.name + '}';
+      }
+
+      return token;
+    })
+    .join('');
+};
+
+/**
+ * @description Builds the required object for a path parameter
+ *
+ * @param {string} routePath - The route's path property
+ *
+ * @returns {object } Swagger path params object
+ */
+const getPathParams = routePath => {
+  return pathToRegexp
+    .parse(routePath)
+    .filter(token => _.isObject(token))
+    .map(param => {
+      return {
+        name: param.name,
+        in: 'path',
+        description: '',
+        deprecated: false,
+        required: true,
+        schema: { type: 'string' },
+      };
+    });
+};
+
+/**
+ *
+ * @param {string} prefix - The prefix found on the routes object
+ * @param {string} path - The current route
+ *
+ * @returns {string}
+ */
+const getPathWithPrefix = (globalRoutePrefix, route) => {
+  if (_.has(route.info, 'pluginName') && _.has(route.config, 'prefix')) {
+    // Use the prefix specified on a plugin route
```

Comment:
@alexandrebodin 
I think I’m a bit confused. The prefixes specified on a plugin route aren’t added to the route path, the default prefix is just removed and I can’t find where this is happening.

It seems like a condition similar to this one needs to be added to the [getPath function here](https://github.com/strapi/strapi/blob/d057976298df63516167d58bca5af5d706e0811e/packages/core/strapi/lib/services/server/compose-endpoint.js#L11)

For example:

The users-permissions user `/users` route is specifying a `prefix: ''` and somehow this replaces the default `prefix: '/users-permissions'`. The route is available at `/api/users` and it is reflected correctly on the users-permissions settings in the admin.

However if I change the prefix to `prefix: '/test'` the route will still be available at `/api/users` but the users-permissions settings in the admin will show `/api/test/users`

For an api the prefix is not displayed correctly in the users-permissions settings in the admin panel. For this I think [the loop on the apis](https://github.com/strapi/strapi/blob/d4b59a1029617303019cbe43ee04b986375b04eb/packages/plugins/users-permissions/server/services/users-permissions.js#L103) in the users-permissions plugin needs to be updated to add a custom api route prefix.